src/cache.rs

```diff
@@ -475,6 +432,60 @@ impl Cache {
     }
 }
 
+fn build_proxy() -> Result<Option<Proxy>> {
+    if let Ok(ref host) = env::var("HTTP_PROXY") {
+        return Ok(Some(Proxy::new(host)?));
+    }
+    if let Ok(ref host) = env::var("HTTPS_PROXY") {
+        return Ok(Some(Proxy::new(host)?));
+    }
+
+    Ok(None)
+}
+
+impl Cache {
+    fn build_client(tls_backend: TlsBackend) -> Result<Agent> {
+        let proxy = build_proxy()?;
+        let mut tls_builder = TlsConfig::builder();
+        tls_builder = match tls_backend {
```

Comment:
Can't we omit the first line and use the following?
```rust
let tls_builder = match tls_backend {    
    TlsBackend::NativeTls => TlsConfig::builder()
        .provider(...)
    // ...
};
```

Reply:
Sure.